GoogleForCreators/web-stories-wp
# Uninstall: Improve logic of uninstall code

## Context



## Summary

Move the logic to delete data on uninstall to service classes. On uninstall, simply run the on_plugin_uninstall method on uninstall plugin. 

## Relevant Technical Choices



## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Create a new site, for example on [Local](https://localwp.com/) environment or a service like tastewp.
2. Install built version of this plugin (see comment below for downloading production zip file). 
3. Activate plugin. 
5. On the settings page, toggle the option remove data on uninstall. 
6. Create some stories. 
7. Upload some media. 
8. Create web stories categories. 
9. Uninstall plugin. 
10. Reinstall plugin. 
11. See all data has been removed. 


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #12361
